### PR TITLE
Fix mobile menu modals not opening for the second time

### LINKED ISSUE #1308: Unable to open modal more than one time per page load
## Description

The mobile content switcher can only be opened once on iOS.

## Reproduction

1. Go to `/search?q=greyhound`
2. Tap the content switcher and either select a different Content Type or tap the x to close and return to the search results.
3. Tap the content switcher a second time; nothing happens, and the content switcher does not open.

## Environment

 - Device: iPhone XS
 - OS: iOS 15.4.1
 - Browser: Chrome
 - Version: 100.0.4896.85
 
Original reporter @panchovm. This could not be reproduced in Firefox on Android.

## Resolution

- [ ] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #1308 by @stacimc

## Description

This PR sets the `useHideOutside` to false for mobile modals. This fixes the bug, but it doesn't fix the underlying problem. That, hopefully, will be fixed in PR that Typescriptifies the dialog composables: https://github.com/WordPress/openverse-frontend/pull/1279

## Testing Instructions

Open the app on mobile width, and try clicking Filter/ Page menu buttons in the header several times. The menus should open and close successfully. On main [search-staging.openverse.engineering/search](https://search-staging.openverse.engineering/search/?q=cat), you can only open them once.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by sarayourfriend** [comment]
LGTM, but could we just add the prop as false to the base modal implementation so that it is fixed across all modals, not just the mobile ones? For example, the modal in storybook still has the problem.

**Review by sarayourfriend — APPROVED** [review]
LGTM! This is even testable with our Storybook preview :tada:

File: src/components/VModal/VModalContent.vue [hunk 1]
```diff
@@ -92,7 +92,7 @@ const VModalContent = defineComponent({
     },
     hideOnClickOutside: {
       type: Boolean,
-      default: true,
+      default: false,
```
Comment (obulat) [inline src/components/VModal/VModalContent.vue:95]: This is the change that fixes the bug, everything else is ... refactoring, I guess :) and tests

**Review by krysal — APPROVED** [review]
Great find! 💯

**Pull request merged**